### ISSUE TITLE
NZBhydra2 info link going to download url

### ISSUE THREAD

**Issue by AmazingYan-1** [Jackett/Jackett#13002]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Info Starting Jackett v0.20.529 
Info Environment version: 6.0.1 (/home/abc/Jackett/) 
Info OS version: Unix 5.4.0.26 (64bit OS) (64bit process) 
Info Jackett variant: CoreLinuxAmdx64 
Info File /etc/issue: Ubuntu 20.04.2 LTS \n \l 
Info Running in Docker: No 
0Info ThreadPool MaxThreads: 32767 workerThreads, 1000 completionPortThreads 
Info App config/log directory: /home/abc/.config/Jackett 
Info Using proxy: Disabled 
Info Using FlareSolverr: No 
Info Using HTTP Client: HttpWebClient2
```

### Description

First of all i thought it's a nzbhydra2 issue and I opened a bug case with them but they asked me to open a bug report with jackett -  https://github.com/theotherp/nzbhydra2/issues/758

My setup is very simple. Nzbhydra2 retrieving records from jackett.

When I search for a torrent using nzbhydra2 via Jackett as an indexer, and when i click the "i" icon next to the download icon, it should take me to the details page right? 95% its working fine. But so far I've identified when we're using torrentleech and Blutopia as your tracker, that "i" icon is directly going to the download url. like this -

i.e. https://www.torrentleech.org/download/2019509/torrentname.torrent

Using nzbhydra2, simply select Torrentleech or Blutopia as your Indexers and search something. You will see the results. From there select any one of the listing and click the "i" icon. Screenshot - https://www.imagebam.com/view/ME7QNLS

For example in nzbhydra if i compare 2 tracker results -

HD torrents (i) icon is linked to - https://hd-torrents.org/details.php?id=XXXXXXXXXXXXXXXXXXXXXXXXXX (which is correct)

Whereas Torrentleech (i) icon is linked to - https://www.torrentleech.org/download/123123123/XXXXXXX.torrent (which is incorrect because the info link it downloading that .torrent file)

I have around 20 indexers and noticing this issue with torrenleech and Blutopia only...

I contacted nzbhydra devs thinking this is an issue with their script but they told me they're just showing whatever is returned by Jackett.. That;s why I'm confused now. 

Let me know if you need more details.

### Logged Error Messages

```text
No errors but this is the log from nzbhydra -

`<html>
<body>
<!--StartFragment-->

Time (newest first) | Level | Logger | Message
-- | -- | -- | --
2022-02-27 14:05 |   | SearchWeb | Web search took 2336ms
2022-02-27 14:05 |   | Searcher | Returning results 1-12 from 12 results in cache. A total of 15 results is available from indexers of which 3 were already rejected
2022-02-27 14:05 |   | Torznab | TorrentLeech: Successfully executed search call in 2272ms with 15 total results
2022-02-27 14:05 |   | SearchResultAcceptor | Rejected 3 search results from TorrentLeech for the following reason: Not enough seeders
2022-02-27 14:05 |   | Torznab | TorrentLeech: Calling http://127.0.0.1:9117/api/v2.0/indexers/torrentleech/results/torznab/api?apikey=XXXXXXXXXXXXXXXXXXXXXXX&t=search&extended=1&q=abc%20def%20home&password=1&limit=1000
2022-02-27 14:05 |   | IndexerForSearchSelector | Selected 1 out of 26 indexers: TorrentLeech

<!--EndFragment-->
</body>
</html>`
```

### Screenshots

![Capture](https://user-images.githubusercontent.com/98995500/155870988-ccf695dd-d442-4aa6-ad7c-86d5ed477cd6.JPG)

**Comment by garfield69** [comment]
Confirmed.
As you may know, Jackett recently added a new feature to its Cardigann processor so it could handle torrent sites that use a JSON API.
I've cross checked a few indexers and I can see that all the indexers that use the Jackett Cardigann JSON API (for example blutopia, torrentleech) return results with an item field called `guid` loaded with the torrent download link, while other indexers using the older Jackett Cardigann HTML (for example alpharatio, iptorrents) return results with the `guid` loaded with the torrent details page link.

I will update the code so that the JSON API indexers will also return the details link in the `guid` field.

**Comment by garfield69** [comment]
Ah. More complicated than I first assumed.

TLDR; NZBhydra2 should not be using the torznab GUID field as the info link, but the COMMENTS field.

Details:
Coincidentally, you picked indexers that are using the Cardigann  JSON API, but the issue goes back deeper.
Keep in mind that there are two classes of indexer, those written in C# and those written in YAML (which get processed by Jackett Cardigann module).

Firstly, the GUID is supposed to represent a unique id for every single torrent in the results from an indexer.
This is used by apps to ignore duplicate torrents, and by Jackett in caches to avoid duplicates too.

It used to be the case that most torrent web sites had a one-torrent one-details page organisation, and most of the Jackett C# indexers would load the details page link into the GUID field, since this was sufficient to identify a torrent.
As time went on however, some torrent sites started presenting results in group presentation, which is to say, there would be a common title, and related torrents listed below.
The details page for all the related torrents in a group would point to the same group title, so the GUID would no longer be unique for each torrent, since they all used the same details page link.
So those C# indexers that deal with torrent groups load the download link into the GUID since this is guaranteed to be unique for each torrent/magnet.
And the Cardigann processor now does the same.
The majority of C# indexer that correspond to web sites that still have a one-torrent one-details structure have not been changed yet, since there is no issue with torrent id duplication and there is no urgent need to update them.

Basically the underlying issue here is that NZBhydra2 has been using the GUID as the torrent info link to take you to the website's torrent detail page, but this was never its purpose.
The correct torznab field to use when wanting to visit the torrent details page on a web site is the COMMENTS link.

**Comment by AmazingYan-1** [comment]
thanks so much for the detailed explanation @garfield69 I really appreciate it and I;m sure this info will help others as well.
